src/main/java/frc/robot/subsystems/drive/DriveSubsystem.java

```diff
@@ -0,0 +1,617 @@
+  }
+
+  public PathPlannerPath pathToPoint(Pose2d finalPose, double finalSpeedMetersPerSec) {
+    Pose2d curPose = getPose();
+    Transform2d finalTransform = new Transform2d(finalPose.getTranslation(), finalPose.getRotation());
+    System.out.println(
+        "Trajectory Transform: " + finalTransform.getX() + " " + finalTransform.getY());
+    Rotation2d finalHeading = Rotation2d.fromDegrees(180);
+    Rotation2d finalHolonomicRotation = finalPose.getRotation();
+
+    PathPlannerPath path = PathPlannerPath.fromPathPoints(Arrays.asList(new PathPoint(finalTransform.getTranslation(),
+        new RotationTarget(Constants.PLACEHOLDER_DOUBLE, finalHolonomicRotation))),
+        new PathConstraints(DriveConstants.DRIVE_WHEEL_FREE_SPEED_METERS_PER_SECOND,
+            DriveConstants.DRIVE_WHEEL_MAX_ACCEL_METERS_PER_SEC_SQRD, DriveConstants.DRIVING_MOTOR_FREE_SPEED_RPS,
+            DriveConstants.DRIVING_MOTOR_MAX_ACCEL_RPS_SQRD),
+        new GoalEndState(finalSpeedMetersPerSec, finalHolonomicRotation));
+
+    return path;
+  }
+
+  // TODO check this
+  public Optional<PathPlannerPath> poseToPath() {
+    Pose2d curPose = getPose();
+    double coastLatencySec = 0.00;
+    Transform2d coastTransform = new Transform2d(
+        new Translation2d(
+            lastSetChassisSpeeds.vxMetersPerSecond * coastLatencySec,
+            lastSetChassisSpeeds.vyMetersPerSecond * coastLatencySec),
+        Rotation2d.fromRadians(lastSetChassisSpeeds.omegaRadiansPerSecond * coastLatencySec));
+    Pose2d futurePose = curPose.plus(coastTransform);
+
+    System.out.println("Acquired target? " + targetPose.isPresent());
+    if (!targetPose.isPresent()) {
+      return Optional.empty();
+    }
+
+    Pose2d finalPose = targetPose.get();
+    Transform2d finalTransform = new Transform2d(finalPose.getTranslation(), finalPose.getRotation());
+    Rotation2d finalHeading = Rotation2d.fromDegrees(180);
+    Rotation2d finalHolonomicRotation = Rotation2d.fromDegrees(0);
+
+    PathPlannerPath path = PathPlannerPath.fromPathPoints(Arrays.asList(new PathPoint(finalTransform.getTranslation(),
+        new RotationTarget(Constants.PLACEHOLDER_DOUBLE, finalHolonomicRotation))),
+        new PathConstraints(DriveConstants.SLOW_LINEAR_SPEED_METERS_PER_SEC,
+            DriveConstants.SLOW_LINEAR_ACCELERATION_METERS_PER_SEC_SQ, DriveConstants.SLOW_ANGULAR_SPEED_RAD_PER_SEC,
+            DriveConstants.SLOW_ANGULAR_ACCELERATION_RAD_PER_SEC_SQ),
+        null);
+    boolean generatedPath = true;
+    return Optional.of(path);
+  }
+
+  /**
+   * Driving inputs will get multiplied by the throttle value, so it should be in
+   * [0,1]
+   */
+  public void throttle(double throttleValue) {
+    throttleMultiplier = throttleValue;
+  }
+
+  public void offsetCurrentHeading(double offsetDegrees) {
+    targetHeadingDegrees = getHeadingDegrees() + offsetDegrees;
+  }
+
+  public void setZeroTargetHeading() {
+    targetHeadingDegrees = 0.0;
+  }
```

Comment:
We may need to swap this out for just "pointing forward" which will flip directions depending on color